infra/gravity/node_commands.go

```diff
@@ -142,8 +137,10 @@ type ClusterStatus struct {
 	Application Application `json:"application"`
 	// Cluster is the name of the cluster
 	Cluster string `json:"domain"`
-	// Status is the cluster status
-	Status string `json:"state"`
+	// State is the cluster state
+	State string `json:"state"`
+	// SystemStatus is the cluster status, see https://github.com/gravitational/satellite/blob/7.1.0/agent/proto/agentpb/agent.proto#L50-L54
```

Comment:
With the bits of 7.0.x being copy-pasted here and there - have you considered vendoring from the gravity repository directly?

Reply:
> have you considered vendoring from the gravity repository directly?

TLDR; Yes. It is complex to do so and didn't seem high ROI.

I half implemented  (got a working build with dep on 7.0) this solution before breaking off and returning to the preexisting "copy key elements" strategy.

My concerns are:

1) Gravity core isn't developed with the library use case in mind.

I don't particularly want to push for this expectation -- I don't see it as an effective use of our time or limited review focus to start considering this now. @r0mant talked a bit about this back in January, when I first stumbled on https://github.com/golang/go/issues/32945:

> at least for gravity, I think it would problematic to migrate to v-prefixed versions
> well, maybe not problematic but will certainly requires code changes for little gain
> but then again, this is probably more important for libraries you vendor (which gravity is not)
> (so things like trace/roundtrip/satellite and other libs we use)

2) I'm concerned  formal dependencies will create a loop or obtuse case in our dependency graph.

Gravity doesn't presently have robotest as a go dependency or vice versa.  However there is an docker container dependency from gravity -> robotest.  And in https://github.com/gravitational/robotest/pull/229#discussion_r429019626 I explored the idea of having Gravity's tests depend explicitly on Robotest (via native golang API calls -- basically moving most of `suite/*` to the gravity repo).

Consider:

Gravity X (5.5? 8.0?) tests depends on robotest Y.  Robotest Y depends on Gravity Z (7.0).  How do we keep  X and Z sane?  It is possible, but would take further design than a naive "robotest as used today depends on gravity".

When these values are redeclared we have the possibility to have `case` statements to appropriately branch based on gravity feature detection.

3) I'd like to avoid the situation where we need to maintain & port to a separate branch of robotest for each active release in gravity.

I know this is where planet & satellite have ended up.  However, because Gravity is test code: if we get to the point where each version of gravity needs a custom version of robotest, I'd prefer to move the gravity version specific parts of robotest into the gravity repo, where they'll be managed more like unit tests.  There may still remain some core (e.g. provisioning) that lives on as a library in this repo.  This will allow for the tight dependencies you're suggesting, without the overhead of "a patch for gravity" -> "a patch for robotest" -> "a patch for gravity" whenever gravity breaks robotest's expectations.

Let me know if you have questions about this.

Reply:
Ok, this definitely sounds like trouble, so it's been cleared.